sceptre/connection_manager.py

```diff
@@ -31,15 +31,20 @@ class ConnectionManager(object):
     _session_lock = threading.Lock()
     _client_lock = threading.Lock()
 
-    def __init__(self, region, iam_role=None):
+    def __init__(self, region, iam_role=None, iam_profile=None):
         self.logger = logging.getLogger(__name__)
-
+        self.exception = None
         self.region = region
+        self.iam_profile = iam_profile
         self.iam_role = iam_role
         self._boto_session = None
 
         self.clients = {}
 
+        if(self.iam_role is not None and self.iam_profile is not None):
+            self.exception = "iam_role and iam_profile are mutually exclusive"
```

Comment:
Not sure this is true - you could use a profile to set your base IAM User, then sceptre would then assume an IAM Role stated.